### ISSUE TITLE
Investigate how to speed training while maintaining rollout skill

### ISSUE THREAD

**Issue by oliverwm1** [ai2cm/ace#65]
Our iteration time is a bit slow. Currently our default setup trains on 150 year-epochs of data (1 year x 150 epochs) with some key parameters being embed_dim=764 and patch_size=1. Using 4 GPUs, this leads to a total training time of about 35 hours ([example](https://wandb.ai/ai2cm/fourcastnet-era5/runs/eim4vfwq/overview)). We want to be able to do a 'complete' training job overnight (<16 hours).

Some possible changes we could make to training:
1. Use more GPUs (sometimes possible, but usually each node on general-cirrascale has at most ~6 GPUs available and often less; we could ask for access to the general A100 cluster; could also run jobs with 8 GPU on other team's clusters in preemptible mode)
2. Use larger patch size
3. Use smaller embed_dim
4. Train for fewer epochs

Previous work indicated that the current parameters gave best performance (e.g. see [wandb group](https://wandb.ai/ai2cm/fourcastnet-era5/groups/fv3gfs-patch-sensitivity/workspace?workspace=user-oliverwm)) but perhaps we should revisit.

**Comment by oliverwm1** [comment]
Another way of saying this: "do a hyperparameter sweep with constraints such that the GPU-hours required for training is below a given threshold"

**Comment by mcgibbon** [comment]
We may be able to achieve faster training through two additional methods: a) reducing horizontal resolution or b) eliminating the annual cycle. The latter can be achieved without editing model code by conducting an initial condition ensemble of 1-month simulations during the same month and using the results for training. Testing the model on the same initial condition with a different perturbation (or even better, ensemble of pertubations) could serve as a simple test which is likely to fail (have compounding biases) at this stage.

**Comment by oliverwm1** [comment]
Worth noting that training SFNO is quite a bit faster. For `sfno-on-fv3gfs-nvidia-params-sf1` [link](https://wandb.ai/ai2cm/fourcastnet-era5/runs/12om7972/overview?workspace=user-oliverwm), each epoch of training takes about 2.8 GPU-hours while `afno-training-fv3` [link](https://wandb.ai/ai2cm/fourcastnet-era5/runs/7bps6245/logs?workspace=user-oliverwm) takes about 7 GPU-hours. In both cases, an epoch is 10 years of data. And if we just want to get a 'decently' good validation loss (say, 0.15 for this case) we can get away with fewer epochs for SFNO. However, as shown in figure below, AFNO does get lower validation loss after 14 epochs.

Plot:

![Screen Shot 2023-04-26 at 3 33 58 PM](https://user-images.githubusercontent.com/23640870/234717228-8d319426-f7da-4e26-9012-98b9a84ad050.png)

**Comment by oliverwm1** [comment]
This issue is out of date, given the significant speed ups we've gotten from moving to the latest SFNO architecture (e.g. https://docs.google.com/presentation/d/1oRnLBsTgAxYj2tJ6C9xHNh0MIX7buISnnw_9hJ5jqTA/edit#slide=id.g24d13cff1a3_0_6).

We can discuss other possible strategies (horizontal coarsening or different dataset) in a separate issue if need be.
